### PR TITLE
Improves weather-symbols classification for sleet

### PR DESCRIPTION
**Pull request by MoseleyS**
IMPRO-1641 wxsym better sleet

Allows more flexibility in coding in multivariate conditions so that we can specify that sleet is when both of these are true:

- Prob(Rain) < Prob(not Rain)
- Prob(Snow) < Prob(not Snow)

Adds handling for unset data points:
- Initial value is -1
- Any values in the final data set that do not fall in the list of valid codes will trigger a ValueError

Testing:
 - [x] Ran tests and they passed OK
 - [x] Added new tests for the new feature(s)

### CODE REVIEW

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- improver_tests/wxcode/wxcode/test_WeatherSymbols.py  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5334158&bid=17459791)

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- improver_tests/wxcode/wxcode/__init__.py  1
         

Clones added
============
- improver_tests/wxcode/wxcode/test_WeatherSymbols.py  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5334158&bid=17459791)

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- improver_tests/wxcode/wxcode/__init__.py  1
         

Clones added
============
- improver_tests/wxcode/wxcode/test_WeatherSymbols.py  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5334158&bid=17459791)

File: improver_tests/wxcode/wxcode/test_wxcode_decision_tree.py [hunk 1]
```diff
@@ -37,6 +37,7 @@
 from improver.wxcode.wxcode_decision_tree_global import (
     START_NODE_GLOBAL, wxcode_decision_tree_global)
 from . import check_diagnostic_lists_consistency
+from improver.wxcode.weather_symbols import WeatherSymbols
```
Comment (benfitzpatrick) [inline improver_tests/wxcode/wxcode/test_wxcode_decision_tree.py:40]: Codacy Issue found: [first party import "from improver.wxcode.weather_symbols import WeatherSymbols" should be placed before "from . import check_diagnostic_lists_consistency"](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5334158)

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- improver_tests/wxcode/wxcode/__init__.py  1
         

Clones added
============
- improver_tests/wxcode/wxcode/test_WeatherSymbols.py  1
         

Clones removed
==============
+ improver/wxcode/wxcode_decision_tree_global.py  -7
+ improver/wxcode/wxcode_decision_tree.py  -7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5334158&bid=17459791)

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- improver_tests/wxcode/wxcode/__init__.py  1
         

Clones added
============
- improver_tests/wxcode/wxcode/test_WeatherSymbols.py  1
         

Clones removed
==============
+ improver/wxcode/wxcode_decision_tree.py  -7
+ improver/wxcode/wxcode_decision_tree_global.py  -7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5334158&bid=17459791)

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- improver_tests/wxcode/wxcode/__init__.py  1
         

Clones added
============
- improver_tests/wxcode/wxcode/test_WeatherSymbols.py  1
         

Clones removed
==============
+ improver/wxcode/wxcode_decision_tree_global.py  -7
+ improver/wxcode/wxcode_decision_tree.py  -7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5334158&bid=17459791)

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- improver_tests/wxcode/wxcode/__init__.py  1
         

Clones added
============
- improver_tests/wxcode/wxcode/test_WeatherSymbols.py  1
         

Clones removed
==============
+ improver/wxcode/wxcode_decision_tree.py  -7
+ improver/wxcode/wxcode_decision_tree_global.py  -7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5334158&bid=17459791)

File: improver/wxcode/wxcode_decision_tree_global.py [hunk 2]
```diff
@@ -53,13 +53,9 @@ def wxcode_decision_tree_global():
...
         * diagnostics_fields: The diagnostics which are being used in the
-              query. If this is a list of lists, the two fields in a given
-              list are subtracted (1st - (2nd * gamma)) and then compared
+              query. If this is a list of lists, the fields are manipulated
+              according to the operators in the list and then compared
               with the probability threshold.
-        * diagnostic_gamma (NOT UNIVERSAL): This is the gamma factor that
-              is used when comparing two fields directly, rather than
-              comparing a single field to a probability threshold.
-              e.g. gamma * P(SnowfallRate) < P(RainfallRate).
```
Comment (bayliffe) [inline improver/wxcode/wxcode_decision_tree_global.py:62]: As we don't define these gammas anymore, the ```0.7``` value seems to appear without explanation in various places. Could you document them somewhere, either this doc-string or with inline comments near to where they occur, just to explain what they are doing. The original explanation that's been removed wasn't great anyway, not really explaining their purpose, so there is room to improve.

File: improver_tests/wxcode/wxcode/test_WeatherSymbols.py [hunk 3]
```diff
@@ -395,7 +415,60 @@ def test_old_naming_convention(self):
...
+            [AuxCoord(0.1, units='mm hr-1'),
+             AuxCoord(0.1, units='mm hr-1')]]
+        query['rain_or_snow']['diagnostic_conditions'] = [
+            ['above', 'above'],
+            ['above', 'above'],
+        ]
+        plugin = WeatherSymbols()
+        test_condition = query['rain_or_snow']
+        result = plugin.create_condition_chain(test_condition)
+        print(result)
```
Comment (bayliffe) [inline improver_tests/wxcode/wxcode/test_WeatherSymbols.py:451]: Can remove the print statement.

File: improver/wxcode/weather_symbols.py [hunk 4]
```diff
@@ -506,9 +564,29 @@ def create_symbol_cube(cubes):
...
+                Weather symbols cube to be checked.
+
+        Raises:
+            ValueError:
+                If any cube.data are not in
+                cube.attributes['weather_code']
+        """
+        if not np.isin(cube.data, cube.attributes['weather_code']).all():
+            raise ValueError(f"Output {cube.name()} cube contains invalid "
+                             "weather codes")
```
[the remark this answers is not in the document]
Comment (bayliffe) [reply improver/wxcode/weather_symbols.py]: You pushed up a change to address this as I was reviewing.

File: improver_tests/wxcode/wxcode/test_WeatherSymbols.py (hunk 3 above)
Comment (MoseleyS) [reply improver_tests/wxcode/wxcode/test_WeatherSymbols.py]: Oops. That's one reason for having a review!

File: improver/wxcode/wxcode_decision_tree_global.py (hunk 2 above)
Comment (MoseleyS) [reply improver/wxcode/wxcode_decision_tree_global.py:62]: Doc-string updated

File: improver/wxcode/weather_symbols.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (MoseleyS) [reply improver/wxcode/weather_symbols.py]: ;-) Yes. The latest change has replaced this with something better.

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- improver_tests/wxcode/wxcode/__init__.py  1
         

Clones added
============
- improver_tests/wxcode/wxcode/test_WeatherSymbols.py  1
         

Clones removed
==============
+ improver/wxcode/wxcode_decision_tree_global.py  -7
+ improver/wxcode/wxcode_decision_tree.py  -7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5334158&bid=17459791)

File: improver/wxcode/wxcode_decision_tree_global.py (hunk 2 above)
Comment (bayliffe) [reply improver/wxcode/wxcode_decision_tree_global.py:62]: Thanks Stephen.

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- improver_tests/wxcode/wxcode/__init__.py  1
         

Clones added
============
- improver_tests/wxcode/wxcode/test_WeatherSymbols.py  1
         

Clones removed
==============
+ improver/wxcode/wxcode_decision_tree.py  -7
+ improver/wxcode/wxcode_decision_tree_global.py  -7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5334158&bid=17459791)

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- improver_tests/wxcode/wxcode/__init__.py  1
         

Clones added
============
- improver_tests/wxcode/wxcode/test_WeatherSymbols.py  1
         

Clones removed
==============
+ improver/wxcode/wxcode_decision_tree.py  -7
+ improver/wxcode/wxcode_decision_tree_global.py  -7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5334158&bid=17459791)

File: improver/wxcode/wxcode_decision_tree_global.py [hunk 5]
```diff
@@ -53,13 +53,12 @@ def wxcode_decision_tree_global():
...
-              query. If this is a list of lists, the two fields in a given
-              list are subtracted (1st - (2nd * gamma)) and then compared
-              with the probability threshold.
-        * diagnostic_gamma (NOT UNIVERSAL): This is the gamma factor that
-              is used when comparing two fields directly, rather than
-              comparing a single field to a probability threshold.
-              e.g. gamma * P(SnowfallRate) < P(RainfallRate).
+              query. If this is a list of lists, the fields are manipulated
+              according to the operators in the list and then compared
+              with the probability threshold. In some cases, a factor of 0.7 is
```
[the remark this answers is not in the document]
Comment (MoseleyS) [reply improver/wxcode/wxcode_decision_tree_global.py:58]: I put it in here to explain its purpose. I could make the language more abstract if you like: "For example, a factor of 0.7 would allow an assymmetric..."

File: improver_tests/wxcode/wxcode/test_utilities.py [hunk 6]
```diff
@@ -511,5 +512,31 @@ def test_raises_exception(self):
...
+        """Test that the get_parameter_names method does what it says."""
+        plugin = WeatherSymbols()
+        condition = ["parameter_name_one", "*", "4.0", "+", "parameter_name_two"]
+        expected = ["parameter_name_one", "parameter_name_two"]
+        result = get_parameter_names(condition)
+        self.assertEqual(result, expected)
+
+    def test_nested(self):
+        """Test getting parameter names from nested lists."""
+        plugin = WeatherSymbols()
```
Comment (benfitzpatrick) [inline improver_tests/wxcode/wxcode/test_utilities.py:528]: Codacy Issue found: [Unused variable 'plugin'](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5334158)

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           

Complexity increasing per file
==============================
- improver_tests/wxcode/wxcode/__init__.py  1
         

Clones added
============
- improver_tests/wxcode/wxcode/test_WeatherSymbols.py  1
         

Clones removed
==============
+ improver/wxcode/wxcode_decision_tree.py  -7
+ improver/wxcode/wxcode_decision_tree_global.py  -7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5334158&bid=17459791)

File: improver/wxcode/weather_symbols.py [hunk 7]
```diff
@@ -256,10 +260,7 @@ def invert_condition(test_conditions):
 
         return inverted_threshold, inverted_combination
 
-    @staticmethod
-    def construct_condition(
-        extract_constraint, condition, probability_threshold, gamma
-    ):
+    def construct_condition(self, extract_constraint, condition, probability_threshold):
```
Comment (benfitzpatrick) [inline improver/wxcode/weather_symbols.py:263]: Codacy Issue found: [Method could be a function](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5334158)

File: improver_tests/wxcode/wxcode/test_utilities.py (hunk 6 above)
Comment (benfitzpatrick) [inline improver_tests/wxcode/wxcode/test_utilities.py:528]: Codacy Issue found: [Undefined variable 'WeatherSymbols'](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5334158)

File: improver/wxcode/weather_symbols.py [hunk 8]
```diff
@@ -105,6 +105,29 @@ def __repr__(self):
         return '<WeatherSymbols tree={} start_node={}>'.format(self.wxtree,
                                                                self.start_node)
 
+    @staticmethod
+    def _is_variable(thing):
```
[the remark this answers is not in the document]
Comment (MoseleyS) [reply improver/wxcode/weather_symbols.py]: I have fixed the CLI help string. I hadn't realised it gave this output, or that there was no unit-test covering it (there still isn't). I'm reluctant to specify that "probaility_of_" must be in all the parameter names as it makes it more difficult to change the weather tree. It is conceivable that a percentile (e.g. median) or deterministic value could be wanted to inform a decision node.

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- improver_tests/wxcode/wxcode/__init__.py  1
         

Clones added
============
- improver_tests/wxcode/wxcode/test_WeatherSymbols.py  1
         

Clones removed
==============
+ improver/wxcode/wxcode_decision_tree_global.py  -7
+ improver/wxcode/wxcode_decision_tree.py  -7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5334158&bid=17459791)

File: improver/wxcode/weather_symbols.py (hunk 7 above)
Comment (benfitzpatrick) [inline improver/wxcode/weather_symbols.py:263]: Codacy Issue found: [Method could be a function](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5334158)

File: improver_tests/wxcode/wxcode/test_wxcode_decision_tree.py [hunk 9]
```diff
@@ -32,6 +32,8 @@
 
 import pytest
 
+from improver.wxcode.utilities import WX_DICT, get_parameter_names
+from improver.wxcode.weather_symbols import WeatherSymbols
```
Comment (benfitzpatrick) [inline improver_tests/wxcode/wxcode/test_wxcode_decision_tree.py:36]: Codacy Issue found: [Unused WeatherSymbols imported from improver.wxcode.weather_symbols](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5334158)

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- improver_tests/wxcode/wxcode/__init__.py  1
         

Clones added
============
- improver_tests/wxcode/wxcode/test_WeatherSymbols.py  1
         

Clones removed
==============
+ improver/wxcode/wxcode_decision_tree.py  -7
+ improver/wxcode/wxcode_decision_tree_global.py  -7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5334158&bid=17459791)

File: improver/wxcode/wxcode_decision_tree_global.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (fionaRust) [reply improver/wxcode/wxcode_decision_tree_global.py:58]: I'm not sure it's worth worrying about. Happy to leave it as it is

File: improver/wxcode/weather_symbols.py (hunk 8 above)
[the remark this answers is not in the document]
Comment (fionaRust) [reply improver/wxcode/weather_symbols.py]: I guess before it probably wouldn't have been picked up by a help CLI test, which I don't think we have any more since moving to pytest for CLI tsets. Do we need to add unit tests for this? I'm not sure they would have helped catch it in this case.

[the remark this answers is not in the document]
Comment (MoseleyS) [reply improver/wxcode/weather_symbols.py]: I have added a pytest which would catch this error in the future.

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- improver_tests/wxcode/wxcode/__init__.py  1
         

Clones added
============
- improver_tests/wxcode/wxcode/test_WeatherSymbols.py  1
         

Clones removed
==============
+ improver/wxcode/wxcode_decision_tree.py  -7
+ improver/wxcode/wxcode_decision_tree_global.py  -7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5334158&bid=17459791)

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- improver_tests/wxcode/wxcode/__init__.py  1
         

Clones added
============
- improver_tests/wxcode/wxcode/test_WeatherSymbols.py  1
         

Clones removed
==============
+ improver/wxcode/wxcode_decision_tree.py  -7
+ improver/wxcode/wxcode_decision_tree_global.py  -7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5334158&bid=17459791)

File: improver/wxcode/weather_symbols.py [hunk 10]
```diff
@@ -358,20 +357,32 @@ def create_condition_chain(self, test_conditions):
...
 
-            gamma = test_conditions.get("diagnostic_gamma")
-            if gamma is not None:
-                gamma = gamma[loop]
             loop += 1
 
-            extract_constraint = self.construct_extract_constraint(
-                diagnostic, d_threshold, self.coord_named_threshold
-            )
+            if isinstance(diagnostic, list):
```
Comment (fionaRust) [inline improver/wxcode/weather_symbols.py:362]: A few comments in this bit of code would be useful.

File: improver_tests/wxcode/wxcode/test_WeatherSymbols.py [hunk 11]
```diff
@@ -847,82 +904,19 @@ def test_weather_data(self):
...
-                1.0,
-                1.0,
-                1.0,
-                1.0,
-                0.0,
-                0.0,
-                0.0,
-                0.0,
-                1.0,
+                [[1.0, 1.0, 1.0], [1.0, 1.0, 1.0], [1.0, 1.0, 1.0]],
```
Comment (fionaRust) [inline improver_tests/wxcode/wxcode/test_WeatherSymbols.py:907]: Could you also do this with the big 1D array at the top of the diff for this file?

File: improver/wxcode/wxcode_decision_tree.py [hunk 12]
```diff
@@ -129,86 +125,88 @@ def wxcode_decision_tree():
...
-            "succeed": 18,
-            "fail": "heavy_rain_or_snow_continuous",
-            "probability_thresholds": [0.0, 0.0],
-            "threshold_condition": ">",
-            "condition_combination": "AND",
+        "heavy_snow_continuous": {
+            "succeed": 27,
+            "fail": "heavy_rain_or_sleet_continuous",
+            "probability_thresholds": [0.0],
+            "threshold_condition": "<",
```
Comment (fionaRust) [inline improver/wxcode/wxcode_decision_tree.py:132]: I'm not sure this change in behaviour is picked up by a unit test (although it is by the acceptance tests). Is that intentional?

File: improver/wxcode/weather_symbols.py [hunk 13]
```diff
@@ -546,7 +558,7 @@ def create_symbol_cube(cubes):
             template_cube,
             attributes,
             optional_attributes=weather_code_attributes(),
-            dtype=np.int32,
+            data=np.ones_like(template_cube.data, dtype=np.int32) * -1,
```
Comment (fionaRust) [inline improver/wxcode/weather_symbols.py:561]: Is this change still needed? I know you discuss it on the ticket,

File: improver_tests/wxcode/wxcode/test_utilities.py [hunk 14]
```diff
@@ -511,5 +507,40 @@ def test_raises_exception(self):
             interrogate_decision_tree("kittens")
 
 
+@pytest.mark.parametrize(
```
Comment (fionaRust) [inline improver_tests/wxcode/wxcode/test_utilities.py:510]: Somehow by adding this test I think you've removed the tests you had for get_parameter_names. I think?

File: improver/wxcode/weather_symbols.py (hunk 13 above)
Comment (MoseleyS) [reply improver/wxcode/weather_symbols.py]: Possibly not, but I prefer having the data array initialised with a missing value of some sort.

File: improver_tests/wxcode/wxcode/test_WeatherSymbols.py (hunk 11 above)
Comment (MoseleyS) [reply improver_tests/wxcode/wxcode/test_WeatherSymbols.py:830]: I've gone through that file and changed ALL the big arrays to be properly structured so that Black formats them nicely.

File: improver/wxcode/weather_symbols.py (hunk 10 above)
Comment (MoseleyS) [reply improver/wxcode/weather_symbols.py:362]: I have added some comments.

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- improver_tests/wxcode/wxcode/__init__.py  1
         

Clones added
============
- improver_tests/wxcode/wxcode/test_WeatherSymbols.py  5
         

Clones removed
==============
+ improver/wxcode/wxcode_decision_tree_global.py  -7
+ improver/wxcode/wxcode_decision_tree.py  -7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5334158&bid=17459791)

File: improver_tests/wxcode/wxcode/test_utilities.py (hunk 14 above)
Comment (MoseleyS) [reply improver_tests/wxcode/wxcode/test_utilities.py:511]: Thanks for pointing this out. I think I probably made a mistake when trying to correct the earlier mistake of committing the change to the wrong branch. I have (hopefully) put these tests back without causing any more problems.

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- improver_tests/wxcode/wxcode/__init__.py  1
         

Clones added
============
- improver_tests/wxcode/wxcode/test_WeatherSymbols.py  5
         

Clones removed
==============
+ improver/wxcode/wxcode_decision_tree_global.py  -7
+ improver/wxcode/wxcode_decision_tree.py  -7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5334158&bid=17459791)

File: improver/wxcode/wxcode_decision_tree.py (hunk 12 above)
Comment (MoseleyS) [reply improver/wxcode/wxcode_decision_tree.py:132]: I suspect that not every pathway through the weather symbols code is tested. I was trying to avoid a wholesale rewrite. How important do you think this is?

File: improver/wxcode/weather_symbols.py (hunk 10 above)
Comment (fionaRust) [reply improver/wxcode/weather_symbols.py:362]: Thank  you

File: improver_tests/wxcode/wxcode/test_WeatherSymbols.py (hunk 11 above)
Comment (fionaRust) [reply improver_tests/wxcode/wxcode/test_WeatherSymbols.py:830]: Thank you :)

File: improver/wxcode/weather_symbols.py (hunk 13 above)
Comment (fionaRust) [reply improver/wxcode/weather_symbols.py]: I think that's reasonable

File: improver_tests/wxcode/wxcode/test_utilities.py (hunk 14 above)
Comment (fionaRust) [reply improver_tests/wxcode/wxcode/test_utilities.py:511]: Yep they have reappeared :)

File: improver/wxcode/wxcode_decision_tree.py (hunk 12 above)
Comment (fionaRust) [reply improver/wxcode/wxcode_decision_tree.py:132]: Yes, I expect you're right. Unless we can find a small tweak to make to one of the points in the existing unit tests, or it's easy to add a simple new test we new data I'm ok to leave it. It is covered by the acceptance test data at the moment, so that's something.
